src/components/TodoPage/TodoPage.tsx

```diff
@@ -0,0 +1,347 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoMain } from '../TodoMain';
+import {
+  addPost,
+  getTodos,
+  updatePost,
+  USER_ID,
+  deletePost,
+} from '../../api/todos';
+import { TodoHeader } from '../TodoHeader';
+import { TodoFooter } from '../TodoFooter';
+import { ErrorNotification } from '../ErrorNotification';
+import { Filter } from '../../types/Enum';
+import { TodoItem } from '../TodoItem';
+
+export const TodoPage: React.FC = () => {
+  const [titleMessage, setTitleMessage] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const inputFocus = useRef<HTMLInputElement>(null);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filtered, setFiltered] = useState<Todo[]>([]);
```

Comment:
You don't need a separate state for filtered todos
Save them in a common const variable and wrap in useMemo with the correct dependencies

General rule: don't create a separate state for a value that can be computed based on other states or props values